src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -742,60 +754,30 @@ public int Seek(ref Vector<byte> byte0Vector, ref Vector<byte> byte1Vector, ref
         }
 
         /// <summary>
-        /// Find first byte
+        /// Locate the first of the found bytes
         /// </summary>
         /// <param  name="byteEquals"></param >
         /// <returns>The first index of the result vector</returns>
-        /// <exception cref="InvalidOperationException">byteEquals = 0</exception>
-        internal static int FindFirstEqualByte(ref Vector<byte> byteEquals)
+        // Force inlining (64 IL bytes, 91 bytes asm) Issue: https://github.com/dotnet/coreclr/issues/7386
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int LocateFirstFoundByte(ref Vector<byte> byteEquals)
         {
-            if (!BitConverter.IsLittleEndian) return FindFirstEqualByteSlow(ref byteEquals);
-
-            // Quasi-tree search
             var vector64 = Vector.AsVectorInt64(byteEquals);
```

Comment:
I'm not sure what you mean. I tried the following change in a local copy of your branch and all the tests still passed.

```diff
diff --git a/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs b/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs
index 4cceb50..efcbe02 100644
--- a/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs
+++ b/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs
@@ -774,10 +774,10 @@ public int Seek(byte byte0, byte byte1, byte byte2)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static int LocateFirstFoundByte(Vector<byte> byteEquals)
         {
-            var vector64 = Vector.AsVectorInt64(byteEquals);
-            long longValue = 0;
+            var vector64 = Vector.AsVectorUInt64(byteEquals);
+            ulong longValue = 0;
             var i = 0;
-            for (; i < Vector<long>.Count; i++)
+            for (; i < Vector<ulong>.Count; i++)
             {
                 longValue = vector64[i];
                 if (longValue == 0) continue;
@@ -785,7 +785,7 @@ internal static int LocateFirstFoundByte(Vector<byte> byteEquals)
             }
 
             // Flag least significant power of two bit
-            var powerOfTwoFlag = (ulong)(longValue ^ (longValue - 1));
+            var powerOfTwoFlag = longValue ^ (longValue - 1);
             // Shift all powers of two into the high byte and extract
             var foundByteIndex = (int)((powerOfTwoFlag * _xorPowerOfTwoToHighByte) >> 57);
             // Single LEA instruction with jitted const (using function result)
```